packages/css/src/input.css

```diff
@@ -2,21 +2,19 @@
   margin: 0; /* Reset native margin, but resepect parent selevtor `.someting > * { margin: ... }` */
 }
 .ds-input {
-  --dsc-input-background--checked: var(--dsc-input-border-color--checked);
-  --dsc-input-background--invalid: var(--dsc-input-border-color--invalid);
   --dsc-input-background--readonly: var(--ds-color-neutral-background-tinted);
   --dsc-input-background: var(--ds-color-neutral-background-default);
-  --dsc-input-border-color--checked: var(--ds-color-base-default);
-  --dsc-input-border-color--invalid: var(--ds-color-danger-border-strong);
   --dsc-input-border-color--readonly: var(--ds-color-neutral-border-subtle);
   --dsc-input-border-color: var(--ds-color-neutral-border-default);
   --dsc-input-border-width--toggle: max(1px, calc(var(--ds-size-1) / 2)); /* Allow border-width to grow with font-size */
   --dsc-input-border-width: 1px;
   --dsc-input-border-style: solid;
-  --dsc-input-color--checked: var(--ds-color-base-contrast-default);
-  --dsc-input-color--invalid: var(--ds-color-danger-base-contrast-default);
-  --dsc-input-color--readonly: var(--ds-color-neutral-text-subtle);
   --dsc-input-color: var(--ds-color-neutral-text-default);
+  --dsc-input-color--readonly: var(--ds-color-neutral-text-subtle);
+  --dsc-input-color--checked: var(--ds-color-base-default);
+  --dsc-input-color--checked-contrast: var(--ds-color-base-contrast-default);
+  --dsc-input-color--invalid: var(--ds-color-danger-text-subtle);
+  --dsc-input-color--invalid-contrast: var(--ds-color-danger-base-contrast-default);
```

Comment:
Since this is still in review, I'll add a comment here.
Using `--dsc-input-color` means that it **should** be used for the css property `color:`.
`color` is for text, but here we are mixing a lot of different tokens, base, contrast, text.. This is using our tokens wrong.
We need **a good** explanation for this

Reply:
yeah, not sold on it completely either. Work in progress :)

Reply:
No, I want chromatics help 😩

Reply:
We use `color` for both text and icons - everything content really. If anything, we should rename our token to `--ds-color-danger-content-subtle` thought this is a bit unconventional. A designer should _always_ ensure sufficient contrast between text and background. Therefore, it is much better to use `-text` variables than `-border` as no designer will (noor need to) verify border color contrast :)
I suggest we stick to a semantic meaning and a11y safety, and especially when it comes to what variables we _consume_ ☺️